src/compat/stdin.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#ifdef WIN32
+#include <windows.h>    // for SetStdinEcho()
+#else
+#include <termios.h>    // for SetStdinEcho()
+#include <unistd.h>     // for SetStdinEcho()
+#include <sys/poll.h>   // for StdinReady()
+#endif
+
+#include <compat/stdin.h>
+
+// https://stackoverflow.com/questions/1413445/reading-a-password-from-stdcin
+void SetStdinEcho(bool enable)
+{
+#ifdef WIN32
+    HANDLE hStdin = GetStdHandle(STD_INPUT_HANDLE);
+    DWORD mode;
+    GetConsoleMode(hStdin, &mode);
+    if (!enable) mode &= ~ENABLE_ECHO_INPUT; else mode |= ENABLE_ECHO_INPUT;
+    SetConsoleMode(hStdin, mode);
+#else
+    struct termios tty;
+    tcgetattr(STDIN_FILENO, &tty);
+    if (!enable) tty.c_lflag &= ~ECHO; else tty.c_lflag |= ECHO;
```

Comment:
A negative integer is implicitly converted to unsigned type here. Please make the conversion explicit.

Reply:
Is there a case when this would be a problem?

Reply:
I may have misunderstood what you were saying here; what are you suggesting I do, exactly?